src/main/java/executor/command/CommandConvert.java

```diff
@@ -0,0 +1,194 @@
+            return null;
+        }
+    }
+
+    /**
+     * convertCurrency is a function used to loop through the string version of the json.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @param amount this is amount of currency which the user wants to convert
+     * @return this function returns the converted amount of the currency or else returns null
+     */
+    private Double convertCurrency(String from, String to, Double amount) {
+        try {
+            String json = getTheJson(from,to);
+            if (json != null) {
+                JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
+                String rate = jsonObject.getAsJsonObject("rates").get(to).getAsString();
+                BigDecimal exchangeRate = new BigDecimal(rate);
+                double exRate = exchangeRate.doubleValue();
+                setExchangeRate(exRate);
+                double convertedAmount = exRate * amount;
+                return convertedAmount;
+            }
+        } catch (Exception e) {
+            Ui.dukeSays(e.getMessage());
+            Ui.dukeSays(Ui.LINE);
+            Ui.dukeSays("Please enter in the following format : "
+                    + "convert 2000 /from USD /to EUR");
+        }
+        return null;
+    }
+
+
+    private String getFrom() {
+        return from;
+    }
+
+    private String getTo() {
+        return to;
+    }
+
+    private Double getAmount() {
+        return amount;
+    }
+
+    /**
+     * result basically returns the to.string() version of the output that the user will be shown
+     * @param convertedAmount this is amount which was converted into
+     * @return string of output is returned
+     */
+    private String result(Double convertedAmount) {
+        convertedAmount = roundTheNum(convertedAmount, 2);
+        return "DUKE$$$ has converted " + this.from
+                + " " + roundTheNum(this.amount, 2) + " "
+                + "to" + " "
+                + this.to + " " + convertedAmount + "\n";
+    }
+
+    /**
+     * roundtheNum is basically a function to round the values to a specific number of decimal places.
+     * @param value this is the value which we want to round
+     * @param places number of decimal places
+     * @return the rounded double value is returned as the output
+     */
+    public double roundTheNum(double value, int places) {
+        if (places < 0) {
+            throw new IllegalArgumentException();
+        }
+
+        BigDecimal bd = BigDecimal.valueOf(value);
+        bd = bd.setScale(places, RoundingMode.HALF_UP);
+        return bd.doubleValue();
+    }
+
+    public Double getExchangeRate() {
+        return exchangeRate;
+    }
+
+    private void setExchangeRate(Double exchangeRate) {
+        exchangeRate = roundTheNum(exchangeRate, 3);
+        this.exchangeRate = exchangeRate;
+    }
```

Comment:
Setters and Getters should be public/package-private or else there's no point of them existing. Group all Setters and Getters together at the bottom of the Class.

Reply:
Moved to the bottom of the class